vcversioner.py

```diff
@@ -73,6 +74,14 @@ def find_version(include_dev_version=True, root='%(pwd)s',
         the version number will be ``1.0.post3``. This behavior can be disabled
         by setting this parameter to ``False``.
 
+    :param use_dev_not_post: PEP 440 states that versions created during
+        development cycles should use the ``.dev`` suffix instead of ``.post.``
+        This suffix can be used by setting this parameter to ``True``. This
+        suffix references a corresponding future release, so vcversioner will
+        assume that the future release is a single version higher than the
+        current commit. For example, if the current commit is one revision past
+        the ``2.0.1`` tag, the computed version will be ``2.0.2dev1``.
```

Comment:
Making that configurable would be fine to me. The reason it updates the last digit of any tag is so that it sorts properly if you ever tag ANY higher version.

For example let's say you're currently on release v1.3.4, and you do some work and make a commit. With this option enabled, you'll get v1.3.5dev1. If you then decide to tag v1.3.5 or v1.4.0 or v2.0.0, they'll all properly sort after the dev version.

Now, if the option chose to give you 1.4.0dev1 instead, and you decided to tag a v1.3.5 because it really was just a patch fix, the dev version will sort as "newer".

The neat benefit of updating the last number is that it works with alpha, beta, and RC builds too. For example, let's say you are preparing for a big v2.0.0 release and want to put some alpha builds out there. You tag a v2.0.0a1 build, send it out to some people to test, and get some feedback about things to fix.  You fix the first bug, commit and build, and the dev option gives you a v2.0.0a2dev1 version. Make another change, commit and build, and you get v2.0.0a2dev2. Now you can tag a new alpha v2.0.0a2 and it'll sort correctly, or go ahead and tag v2.0.0, which will also sort correctly.

Overall, I would be happy with the "postX" functionality except that pip treats those versions as "real" releases compared to alpha, beta, rc, and dev "pre-releases", which you can avoid installing by not using the `--pre` flag.